src/jit/lclvars.cpp

```diff
@@ -2459,12 +2467,28 @@ void Compiler::lvaSetStruct(unsigned varNum, CORINFO_CLASS_HANDLE typeHnd, bool
             varDsc->lvExactSize = info.compCompHnd->getHeapClassSize(typeHnd);
         }
 
-        size_t lvSize = varDsc->lvSize();
+        unsigned lvSize = varDsc->lvSize();
         assert((lvSize % TARGET_POINTER_SIZE) ==
                0); // The struct needs to be a multiple of TARGET_POINTER_SIZE bytes for getClassGClayout() to be valid.
         varDsc->lvGcLayout = getAllocator(CMK_LvaTable).allocate<BYTE>(lvSize / TARGET_POINTER_SIZE);
         unsigned  numGCVars;
         var_types simdBaseType = TYP_UNKNOWN;
+        varDsc->lvType         = impNormStructType(typeHnd, varDsc->lvGcLayout, &numGCVars, &simdBaseType);
+
+#if (defined(_TARGET_AMD64_) && defined(_TARGET_WINDOWS_)) || defined(_TARGET_ARM64_)
```

Comment:
Does this need the `#ifdef _TARGET_WINDOWS_` ?
The method  `getArgTypeForStruct ` should properly handle any differences in the calling convention between Windows and Linux.

Reply:
Made the various ifdefs for this (largely) consistent.